api/lib/models/checkout-items/checkout-items-data-mapper-impl.ts

```diff
@@ -0,0 +1,192 @@
+import { IUowOpts } from 'services/database/svc/uow.service';
+import { Inject, Injectable } from 'injection-js';
+import { from } from 'rxjs';
+import { map } from 'rxjs/operators';
+import * as squel from 'squel';
+import { AuthLevel } from 'services/auth/auth-types';
+import { IAcl, IAclPerm } from 'services/auth/RBAC/rbac-types';
+import { MysqlUow } from 'services/database/svc/mysql-uow.service';
+import { Logger } from 'services/logging/logging';
+import { Stream } from 'ts-stream';
+import { GenericDataMapper } from 'services/database/svc/generic-data-mapper'
+import { IDbResult, IDataMapper } from "services/database";
+import { CheckoutItems } from '.';
+import { UidType } from 'JSCommon/common-types';
+import { HttpError, MethodNotImplementedError } from 'JSCommon/errors';
+import tsStream from 'ts-stream';
+import { IActiveHackathonDataMapper } from '../hackathon/active-hackathon';
+
+/**
+ * TODO: Change 'CHECKOUT_DATA' to reference the checkoutObjectDataMapper and add documentation
+ */
+
+@Injectable()
+export class CheckoutItemsDataMapperImpl extends GenericDataMapper
+  implements IAclPerm, IDataMapper<CheckoutItems> {
+  public readonly COUNT: string = 'checkoutItemss:count';
+  public readonly CREATE: string = 'checkoutItems:create';
+  public readonly DELETE: string = 'checkoutItems:delete';
+  // Undefined actions for checkout items data mapper
+  public readonly READ: string;
+  public readonly READ_ALL: string;
+  public readonly UPDATE: string;
+  
+  public tableName = 'CHECKOUT_ITEMS'
+
+  protected pkColumnName: string = 'uid';
+ 
+  constructor(
+    @Inject('IAcl') acl: IAcl,
+    @Inject('MysqlUow') protected readonly sql: MysqlUow,
+    @Inject('IActiveHackathonDataMapper') protected readonly activeHackathonDataMapper: IActiveHackathonDataMapper,
+    @Inject('BunyanLogger') protected readonly logger: Logger,
+  ) {
+    super(acl);
+    super.addRBAC(
+      [this.CREATE, this.UPDATE, this.DELETE],
+      [AuthLevel.TEAM_MEMBER],
+      undefined,
+      [AuthLevel[AuthLevel.VOLUNTEER]],
+    )
+    super.addRBAC(
+      [this.READ, this.READ_ALL],
+      [
+        AuthLevel.TEAM_MEMBER,
+      ],
+    );
+  }
+  public delete(object: CheckoutItems): Promise<IDbResult<void>> {
+    const query = squel.delete({ autoQuoteTableNames: true, autoQuoteFieldNames: true })
+      .from(this.tableName)
+      .where(`${this.pkColumnName} = ?`, CheckoutItems)
```

Comment:
```suggestion
      .where(`${this.pkColumnName} = ?`, object.id)
```
A better suggestion is to use a string as a parameter instead and use that.